src/services/anchor-service.ts

```diff
@@ -546,40 +547,46 @@ export class AnchorService {
    */
   async _persistAnchorResult(anchors: FreshAnchor[], candidates: Candidate[]): Promise<number> {
     // filter to requests for streams that were actually anchored successfully
-    const acceptedRequests = []
+    const acceptedRequests: Request[] = []
     for (const candidate of candidates) {
       acceptedRequests.push(candidate.request)
     }
 
-    const trx = await this.connection.transaction(null, { isolationLevel: 'repeatable read' })
-    try {
-      const persistedAnchorsCount =
-        anchors.length > 0
-          ? await this.anchorRepository.withConnection(trx).createAnchors(anchors)
-          : 0
-
-      await this.requestRepository.withConnection(trx).updateRequests(
-        {
-          status: RS.COMPLETED,
-          message: 'CID successfully anchored.',
-          pinned: true,
+    return await this.connection
+      .transaction(
+        async (trx) => {
+          const persistedAnchorsCount =
+            anchors.length > 0
+              ? await this.anchorRepository.withConnection(trx).createAnchors(anchors)
+              : 0
+
+          await this.requestRepository.withConnection(trx).updateRequests(
+            {
+              status: RS.COMPLETED,
+              message: 'CID successfully anchored.',
+              pinned: true,
+            },
+            acceptedRequests
+          )
+
+          // record some metrics about the timing and count of anchors
+          const completed = new TimeableMetric(SinceField.CREATED_AT)
+          completed.recordAll(acceptedRequests)
+          completed.publishStats(METRIC_NAMES.CREATED_SUCCESS_MS)
+          Metrics.count(METRIC_NAMES.ACCEPTED_REQUESTS, acceptedRequests.length)
+          return persistedAnchorsCount
         },
-        acceptedRequests
+        { isolationLevel: 'repeatable read' }
       )
-
-      await trx.commit()
-
-      // record some metrics about the timing and count of anchors
-      const completed = new TimeableMetric(SinceField.CREATED_AT)
-      completed.recordAll(acceptedRequests)
-      completed.publishStats(METRIC_NAMES.CREATED_SUCCESS_MS)
-      Metrics.count(METRIC_NAMES.ACCEPTED_REQUESTS, acceptedRequests.length)
-
-      return persistedAnchorsCount
-    } catch (err) {
-      await trx.rollback()
-      throw err
-    }
+      .catch(async (err) => {
+        if (err?.code === REPEATED_READ_SERIALIZATION_ERROR) {
```

Comment:
should we still log the error here?

Reply:
I can add a log sure

yes the rollback is handled automatically